### ISSUE TITLE
[Linux] Client doesn't show my Hero Builds at the Guide selection panel

### ISSUE THREAD

**Issue by tukkek** [ValveSoftware/Dota-2#1265]
Hi guys and thank you very much for the quick fix on the segfault yesterday :) I can't be a 100% sure this bug is on Linux only but I took some time to look at the general bug-report forums and reddit too and no one seems to have reported it, so I'm thinking it's fair to assume it's on Linux only since if it was affecting all clients it would surely have been reported already.

Basically I have set up a few guides using this page, as many players do http://www.dota2.com/workshop/builds/overview 

My guides are still there normally and I can still edit them normally. One ID I can provide for my guides is `file id = 857222730447709187`, it it helps any. I can send more too.

Up until yesterday they were working fine. However since the breaking update yesterday was fixed and I got into the game again today, my own guides, which were previously marked as "selected" (default) for my favorite heroes are not only non-selected (default) anymore but also don't show up for me at all. This functionality is extremely useful since without them showing up in the listing, there is no way of manually searching for them and selecting them again. As it stands now, **I have no manner of accessing my guides in-game**, even though they were my default ones as of yesterday, making this clearly a bug.

A lack of guides is not game-breaking but it surely affects me a lot as I (and certainly others too) have complex builds with carefully selected item ordering to complete recipes, give the proper boosts during early/mid/late game, etc, which is just unfeasible to remember by memory during an action-packed match.

Thanks again for the quick fix for yesterday's segfault - even though I couldn't play Dota yesterday, it was amazing to see how fast Valve responded to it here on GitHub! I hope you can fix this fast too but I'll be playing with the default guides until then :) thanks again for the great game!

**Comment by tukkek** [comment]
```
System information:

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Desktop
    No Touch Input Detected
    
Processor Information:
    CPU Vendor:  AuthenticAMD
    CPU Brand:  AMD FX-8320E Eight-Core Processor              
    CPU Family:  0x15
    CPU Model:  0x2
    CPU Stepping:  0x0
    CPU Type:  0x0
    Speed:  3200 Mhz
    8 logical processors
    8 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Supported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    Debian GNU/Linux 9.0 (stretch) (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.9.0-3-amd64
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11903000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime-beta-release_2016-12-05
    
Video Card:
    Driver:  NVIDIA Corporation GeForce GT 710/PCIe/SSE2

    Driver Version:  4.5.0 NVIDIA 375.66
    OpenGL Version: 4.5
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 59 Hz
    VendorID:  0x10de
    DeviceID:  0x128b
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1366 x 768
    Desktop Resolution: 1366 x 768
    Primary Display Size: 16.14" x 9.06"  (18.50" diag)
                                            41.0cm x 23.0cm  (47.0cm diag)
    Primary Bus: PCI Express 8x
    Primary VRAM: 2048 MB
    Supported MSAA Modes:  2x 4x 8x 16x 
    
Sound card:
    Audio device: Realtek ALC887-VD
    
Memory:
    RAM:  3938 Mb
    
Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Steam Controller Cable and Base:  Not set
    Total Hard Disk Space Available:  253416 Mb
    Largest Free Hard Disk Block:  3199 Mb
    VR Headset: None detected
```

**Comment by gdrewb-valve** [comment]
The larger team will take a look, if we hear anything back I'll update.

**Comment by tukkek** [comment]
Thank you once again for the quick response!

**Comment by tukkek** [comment]
I figured this one out. May sound obvious but I had to publish my guides to the Steam Workshop and after that subscribe to them. This got them to show up once again on my "hero guides" panels (on picking phase, hero tab on the home screen, etc) - as all my guides were unpublished up to now.

What strikes me as odd though is that prior to the latest batch of patches I didn't have to do any of this. Well, I guess it only makes sense, after all and the "bug" only affects players who were used to having guides show up without publishing them first. New players creating new guides will have to go through the whole process before they show up so they'll know the entire workflow already.

I also found out today that players on Windows 10 have the same issue as well so this doesn't really belong here.

Thanks again for the quick reply and quick fix for other recent issues! I'm taking the liberty to close this.
